### PR TITLE
Retrieve paintings by artist name

### PR DESCRIPTION
**Pull request by lparth**
This patch enables retrieving paintings by any portion of an artist
name. Multiple artists' paintings can be retrieved if names match.

### CODE REVIEW

**Comment by lparth** [comment]
Also posted a demo video of this feature:

https://www.youtube.com/watch?v=KG0YvvAKf6A

If you need updates/modifications on this patch let me know, and thank you for writing this useful tool.

~parthi

File: wikiart/console.py [hunk 1]
```diff
@@ -33,10 +33,11 @@ def __init__(self):
             description='specify which operation to perform.')
         p_fetch = sp.add_parser('fetch', help='fetch paintings from WikiArt')
         p_fetch.add_argument('--only', type=str, default='all',
-                             choices=('artists', 'paintings', 'all'),
+                             choices=('artists', 'paintings', 'all', 'artist'),
```
Comment (lucasdavid) [inline wikiart/console.py:36]: Remove `artist` here.

Comment (lucasdavid) [inline wikiart/console.py]: Remove artist from here.

File: wikiart/fetcher.py [hunk 2]
```diff
@@ -129,6 +130,24 @@ def fetch_artists(self):
 
         return self
 
+    def fetch_artist(self, artistName):
```
Comment (lucasdavid) [inline wikiart/fetcher.py]: `artist_name`

File: wikiart/fetcher.py [hunk 3]
```diff
@@ -129,6 +130,24 @@ def fetch_artists(self):
...
+
+        self.painting_groups = []
+        show_progress_at = max(1, int(.1 * len(self.artists)))
+
+        for artist in self.artists:
+            if re.search(artistName.lower(), artist['artistName'].lower()):
+                self.painting_groups.append(self.fetch_paintings(artist))
+
+        if not self.painting_groups:
+            raise RuntimeError('Artist Not found. Cannot continue.')
```
Comment (lucasdavid) [inline wikiart/fetcher.py]: I believe `Logger.error('...')` is better than raising an error in these two, as many people without computer science background use this tool and don't really understand/find stack traces useful.

File: wikiart/console.py [hunk 4]
```diff
@@ -33,10 +33,11 @@ def __init__(self):
...
                              help='fetch only artists list, paintings '
                                   'metadata or artists, paintings annotations '
                                   'and copies')
+        p_fetch.add_argument('--artistname', type=str, required=False)
```
Comment (lucasdavid) [inline wikiart/console.py]: I think you could merge this parameter with `--only` (by removing choices from only and letting users do `python wikiart.sh --only gogh`).

File: wikiart/console.py [hunk 5]
```diff
@@ -87,6 +88,9 @@ def fetch(self, args):
             if args.only == 'paintings':
                 f.fetch_all_paintings()
 
+            if args.only == 'artist':
```
Comment (lucasdavid) [inline wikiart/console.py]:
```py
if args.only == 'paintings':
  f.fetch_all_paintings()
elif args.only != 'artists':
  # Looking for a specific painter.
  f.fetch_artist(args.only).copy_everything()
```

**Comment by lucasdavid** [comment]
Thank you very much for the contribution. Just a few considerations before merging.

**Comment by lparth** [comment]
Sure, please let me know your comments and I'll make the changes. Thank you.

**Comment by lparth** [comment]
I dropped the commit that had a demo video link from the PR. It was temporarily added for a presentation. It's not needed for the documentation.

**Review by lucasdavid — CHANGES_REQUESTED** [review]

File: wikiart/console.py (hunk 4 above)
Comment (lparth) [reply wikiart/console.py]: Cool, I'll do that update the PR. Cheers.

**Comment by lparth** [comment]
Thanks for the comments and code. I've pushed (squashed into a single commit) the changes. I think raising an error is better when chaining functions, so I raise ValueError and handle it at the console and dump it to the logger to give a user-friendly experience. Let me know what you think. 

Cheers,
~parthi

**Comment by lucasdavid** [comment]
Great. Thank you once again.

**Comment by lparth** [comment]
That's great. Thanks
~parthi

**Pull request merged**